### DISCUSSION TITLE
SCRIPTER: wcs in a subroutine sends text to webpage twice

### DISCUSSION THREAD

**Discussion by UBWH** [arendst/Tasmota#20336] [Q&A]
SCRIPT Question
I have a Script Web page successfully calling a subroutine to display dynamic text . Problem is ... the text appears twice. What am I doing wrong?

SCRIPT EXTRACTS
```
#sub2
wcs Test 1
wcs Test 2

>w TEST
%=#sub2
```

HTML OUTPUT EXTRACT
```
<div id='l1' name='l1'>Test 1Test 2</div>Test 1Test 2
```

Is there a smarter way to cause #sub2 to output HTML than wcs?

**Comment by gemu2015** [comment]
\>W does call subroutine twice, 1. in the upper "sensor" section and 2. on gui redraw.

select with wm flag, see docs

#sub1
if wm==0 {
wcs Hallo
}

or instead of >W, use >WS and >WM, see docs  (you may not mix >W with the 2 others)

**Comment by UBWH** [comment]
Thanks for your quick reply.

But did not work for me. Here is the code. Problem is that my wm value is not zero

AS RECOMMENDED
```
#sub1(stmp)
if wm==0 {
wcs ['weekday','Power'],
wcs ]);var options={title:'%stmp%'};
}

>w TEST
WM =%wm%
%=#sub1("my title")
```
```
<div id='l1' name='l1'>WM =119.00</div>
```

TRY WITH mw!=0
```
#sub1(stmp)
if wm!=0 {
wcs ['weekday','Power'],
wcs ]);var options={title:'%stmp%'};
}

>w TEST
WM =%wm%
%=#sub1("my title")
```

HTML OUTPUT
```
<div id='l1' name='l1'>WM =119.00['weekday','Power'],]);var options={title:'my title'};</div>
['weekday','Power'],]);var options={title:'my title'};
```

**Reply by gemu2015** [reply]
wm should be zero, test with

#sub1(stmp)
print wm=%wm%
if wm==0 {
wcs ['weekday','Power'],
wcs ]);var options={title:'%stmp%'};
}

to verify i tested with esp8266 and esp32, it works

**Comment by UBWH** [comment]
CONSOLE
I am using ESP32

ONE view of the TEST >w page shows two console entries

```
10:37:09.282 wm=119.00
10:37:09.285 wm=120.00
```
I guess I could use as below ... and it would only display once!
```
#sub1(stmp)
print wm=%wm%
if wm==119 {
wcs ['weekday','Power'],
wcs ]);var options={title:'%stmp%'};
}
```
But - why is wm not zero?

**Reply by gemu2015** [reply]
sorry, i now realized that you use >w  (lower case w) not upper case >W
the wm var has different values there. 
use 119 or 120 here

**Comment by UBWH** [comment]
Thanks for your help!
